docs/nx-cloud/recipes/access-tokens.md

```diff
@@ -43,6 +169,81 @@ jobs:
+     ![define gitlab environment for protected branches](/nx-cloud/recipes/gitlab-new-environment.avif)
+2. In your project, navigate to _Settings_ > _CI/CD_ tab and expand the _Variables_ section.
+3. Click on _Add variable_ and fill in the following information:
+   - Type: _Variable_
+   - Environments: _All_
+   - Visibility: _Masked and hidden_
+   - Flags: uncheck _Protected variable_
+   - Description: "read-only token for nx-cloud"
+   - Key: `NX_CLOUD_ACCESS_TOKEN`
+   - Value: Your `read-only` token from Nx Cloud
+4. Click _Add variable_.
+   ![add read-only nx cloud access token to gitlab](/nx-cloud/recipes/gitlab-variable-settings-readonly.avif)
+5. Click on _Add variable_ again and fill in the following information:
+   - Type: _Variable_
+   - Environments: Your protected environment created in step 1
+   - Visibility: _Masked and hidden_
+   - Flags: check _Protected variable_
+   - Description: "read-write token for nx-cloud"
+   - Key: `NX_CLOUD_ACCESS_TOKEN`
+   - Value: Your `read-write` token from Nx Cloud
+6. Click _Add variable_.
+   ![add read-write nx cloud access token to gitlab](/nx-cloud/recipes/gitlab-variable-settings-readwrite.avif)
+7. Now you should see 2 secrets where 1 is a part of the protected & tagged to the environment and the other is not.
+   ![GitLab project variable configuration screen](/nx-cloud/recipes/gitlab-variable-setting.avif)
+8. Update your pipeline to include steps where you want to write to the nx cache with the correct contexts.
+
+Example usage:
+
+```yaml {% fileName=".gitlab-ci.yml" highlightLines=["2-3"] %}
+<job-name>:
+  environment:
+    name: <environment-name>
+```
+
+{% callout type="check" title="Can't someone change the step environment?" %}
+Since we use the _Protected variable_ flag, CI can only read the variable when running in a protected branch. If a developer tries to edit the steps to run a PR in the environment with the `read-write` token will, then the token will not be populated in CI since their branch is not marked as protected.
+
+Take caution though, if you allow team members to have direct write access to a protected branch, then they could modify the steps to write to the nx cache without having a code review first.
+{%/callout %}
+
+### Jenkins
+
+Jenkins configuration can be quite extensive make each Jenkins instance potentially different from each other. Because of this we can only provide a minimal viable approach, but there can be a multiple way to provide scoped access tokens to your pipelines. The goal is to create two areas within Jenkins, where one is the _protected_ and the other is the _unprotected_. These specifically map to how you deem your branches should have read/write vs read permissions. We recommend making branches that developers cannot directly push to and require a code review to merge to, as the _protected_ branches, and the rest being _unprotected_.
```

Comment:
```suggestion
Jenkins configuration can be quite extensive making each Jenkins instance unique. Because of this we can only provide a minimal viable approach, but there can be multiple ways to provide scoped access tokens to your pipelines. The goal is to create two areas within Jenkins, where one is the _protected_ and the other is the _unprotected_. These specifically map to how you deem your branches should have read/write vs read permissions. We recommend making branches that developers cannot directly push to and require a code review to merge to, as the _protected_ branches, and the rest being _unprotected_.
```